FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinStoreTest.m

```diff
@@ -53,21 +57,23 @@ + (BOOL)createSubDirectory:(NSString *)subDirectoryName;
 
 @interface FIRMessagingCheckinStoreTest : XCTestCase
 
+@property(nonatomic, strong) FIRMessagingCheckinStore *checkinStore;
+@property(nonatomic, strong) FIRMessagingBackupExcludedPlist *plist;
 @end
 
 @implementation FIRMessagingCheckinStoreTest
 
 - (void)setUp {
   [super setUp];
   [FIRMessaging createSubDirectory:kSubDirectoryName];
+  self.checkinStore = [[FIRMessagingCheckinStore alloc] init];
+  self.plist = [[FIRMessagingBackupExcludedPlist alloc] initWithFileName:kFakeCheckinPlistName
+                                                            subDirectory:kSubDirectoryName];
+  self.checkinStore.plist = self.plist;
 }
 
 - (void)tearDown {
-  NSString *path = [self pathForCheckinPlist];
-  if ([[NSFileManager defaultManager] fileExistsAtPath:path]) {
-    NSError *error;
-    [[NSFileManager defaultManager] removeItemAtPath:path error:&error];
-  }
+  [self.plist deleteFile:nil];
```

Comment:
Optionally, you can pass in a nonnull error pointer here and validate that the delete succeeds in tests.

Reply:
This is in teardown, I don't usually assert error there but i might be wrong.